nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-file-authorizer/src/test/java/org/apache/nifi/authorization/FileAuthorizerTest.java

```diff
@@ -0,0 +1,207 @@
+        authorizer.onConfigured(configurationContext);
+    }
+
+    @Test
+    public void testAuthorizedUserAction() throws Exception {
+        writeAuthorizationsFile(primary, AUTHORIZATIONS);
+        authorizer.onConfigured(configurationContext);
+
+        final AuthorizationRequest request = new AuthorizationRequest.Builder().resource(ResourceFactory.getFlowResource()).identity("user-1").action(RequestAction.READ).build();
+        final AuthorizationResult result = authorizer.authorize(request);
+        assertTrue(Result.Approved.equals(result.getResult()));
+    }
+
+    @Test
+    public void testUnauthorizedUser() throws Exception {
+        writeAuthorizationsFile(primary, AUTHORIZATIONS);
+        authorizer.onConfigured(configurationContext);
+
+        final AuthorizationRequest request = new AuthorizationRequest.Builder().resource(ResourceFactory.getFlowResource()).identity("user-2").action(RequestAction.READ).build();
+        final AuthorizationResult result = authorizer.authorize(request);
+        assertFalse(Result.Approved.equals(result.getResult()));
+    }
+
+    @Test
+    public void testUnauthorizedAction() throws Exception {
+        writeAuthorizationsFile(primary, AUTHORIZATIONS);
+        authorizer.onConfigured(configurationContext);
+
+        final AuthorizationRequest request = new AuthorizationRequest.Builder().resource(ResourceFactory.getFlowResource()).identity("user-1").action(RequestAction.WRITE).build();
+        final AuthorizationResult result = authorizer.authorize(request);
+        assertFalse(Result.Approved.equals(result.getResult()));
+    }
+
+    @Test
+    public void testReloadAuthorizations() throws Exception {
+        writeAuthorizationsFile(primary, AUTHORIZATIONS);
+        when(configurationContext.getProperty(Mockito.eq("Reload Interval"))).thenReturn(new StandardPropertyValue("1 sec", null));
+        authorizer.onConfigured(configurationContext);
+
+        // ensure the user currently does not have write access
+        final AuthorizationRequest request = new AuthorizationRequest.Builder().resource(ResourceFactory.getFlowResource()).identity("user-1").action(RequestAction.WRITE).build();
+        AuthorizationResult result = authorizer.authorize(request);
+        assertFalse(Result.Approved.equals(result.getResult()));
+
+        // add write access for the user
+        writeAuthorizationsFile(primary, UPDATED_AUTHORIZATIONS);
+
+        // wait at least one second for the file to be stale
+        Thread.sleep(4000L);
+
+        // ensure the user does have write access now using the same request
+        result = authorizer.authorize(request);
+        assertTrue(Result.Approved.equals(result.getResult()));
+    }
+
+    private static void writeAuthorizationsFile(final File file, final String content) throws Exception {
+        byte[] bytes = content.getBytes(StandardCharsets.UTF_8);
+        FileOutputStream fos = new FileOutputStream(file);
```

Comment:
Should probably use a try-with-resources here:
try (final FileOutputStream fos = new FileOutputStream(file)) {
  fos.write(bytes);
}

Or, alternatively, just simply: Files.write(bytes, file);